### DISCUSSION TITLE
adding devices together

### DISCUSSION THREAD

**Discussion by copekyle** [emporia-vue-local/esphome#74] [Q&A]
I'm just starting out and so I apologize for such a simple question. I have a 240V unbalanced load and so I'm measuring both legs. I want to output the total power of the device by adding the 2 ct measurements together. Does this look right?

Sensor Measurement
```   
      - { phase_id: phase_b, input: "13", power: { name: "Dryer B Power", id: cir13, filters: [ *moving_avg, *pos ] } }
      - { phase_id: phase_a, input: "14", power: { name: "Dryer A Power", id: cir14, filters: [ *moving_avg, *pos ] } }
```
Adding them together
```
  - platform: template
    name: "Dryer Total Power"
    lambda: return id(cir13).state + id(cir14).state;
    update_interval: 1s
    id: dryer_total_power
    unit_of_measurement: "W" 
```
I'm wondering if using the id's removes the filtering that was done when I took the measurement.

**Comment by flaviut** [comment]
> I'm wondering if using the id's removes the filtering

It does not. Your approach is good.

For best accuracy, I would suggest doing all the filtering on the template rather than on each measurement.

I would also suggest adding [`internal: true`](https://esphome.io/components/sensor/index.html) to your "Dryer A/B Power" measurements so that they don't show up in Home Assistant.

**Comment by copekyle** [comment]
My ultimate goal is to use these values in home assistant to graph them as 1 value instead of 2 separate values. I noticed in home assistant that it knows the difference between a power value and a template value. Can I make the final result appear as a power value?

**Reply by flaviut** [reply]
```
    unit_of_measurement: "W" 
```

is what should do that. Do you want to upload some screenshots showing what is wrong?

**Reply by copekyle** [reply]
I've currently only got one Vue 2 installed. I'm adding a second to cover the rest of my circuits and I'm programming blind right now while planning out my install for the end of the week. So trying to get everything right for minimal down time.

**Reply by copekyle** [reply]
With my current installed unit, Home Assistant shows different icons for Power, Energy and Templates automatically. So somehow it knows the difference?

![image](https://user-images.githubusercontent.com/10146057/203422687-fd0b0b0f-df30-478e-9b0e-bcb057ba1e02.png)

**Reply by flaviut** [reply]
Oh, that's just the icon. You can change it in ESPHome ([docs](https://esphome.io/components/sensor/index.html), "icon") or Home Assistant, up to you. Doesn't make any difference in functionality.

**Reply by copekyle** [reply]
Great. I really appreciate your quick responses. So I'm thinking this:
```
      - { phase_id: phase_b, input: "13", power: { name: "Dryer B Power", id: cir13, internal: true } }
      - { phase_id: phase_a, input: "14", power: { name: "Dryer A Power", id: cir14, internal: true } }
```
```
  - platform: template
    name: "Dryer Total Power"
    lambda: return id(cir13).state + id(cir14).state;
    update_interval: 1s
    id: dryer_total_power
    unit_of_measurement: "W"
    filters: [*moving_avg, *pos]
    icon: "mdi:flash"
```

**Reply by flaviut** [reply]
Yup! That looks good to me. Only note is this comment from where `moving_avg` is defined:

```
  - &moving_avg
    # we capture a new sample every 0.24 seconds, so the time can
    # be calculated from the number of samples as n * 0.24.
    sliding_window_moving_average:
      # we average over the past 2.88 seconds
      window_size: 12
      # we push a new value every 1.44 seconds
      send_every: 6
```

I'd suggest changing `update_interval: 1s` to `update_interval: 0.24s`, or just leaving it as-is if you don't mind this data being reported less frequently.

**Reply by copekyle** [reply]
got it. Does that go for all templates? I was just using the "Total Power" template as a base to put this together. Seems all of my templates update once a second?

**Reply by copekyle** [reply]
I think I answered my own question. The filtering component is what requires the 0.24 update interval. It took me a bit to understand that note.

**Comment by copekyle** [comment]
One more question. I'm now setting up the total daily energy calculation and I don't think I'm doing this right.
```
  - { power_id: cir13, platform: total_daily_energy, accuracy_decimals: 0, name: "Dryer B Daily Energy", internal: true }
  - { power_id: cir14, platform: total_daily_energy, accuracy_decimals: 0, name: "Dryer A Daily Energy", internal: true }
 ```

```
  - platform: total_daily_energy
    name: "Dryer Daily Energy"
    lambda: return id(cir13).state + id(cir14).state;
    update_interval: 0.24s
    id: dryer_daily_energy
    unit_of_measurement: "Wh"
    icon: "mdi:lightning-bolt"
```

**Reply by flaviut** [reply]
no need to sum them individually:

```
  - { power_id: dryer_total_power, platform: total_daily_energy, accuracy_decimals: 0, name: "Dryer Daily Energy" }
```
